java-frontend/src/main/java/org/sonar/java/bytecode/cfg/Instruction.java

```diff
@@ -174,4 +174,45 @@ public int hashCode() {
       return Objects.hash(super.hashCode(), dim);
     }
   }
+
+  public static class InvokeDynamicInsn extends Instruction {
+
+    private final String desc;
+
+    public InvokeDynamicInsn(String desc) {
+      super(Opcodes.INVOKEDYNAMIC);
+      this.desc = desc;
+    }
+
+    @Override
+    public int arity() {
+      Type methodType = Type.getMethodType(desc);
+      return methodType.getArgumentTypes().length;
+    }
+
+    @Override
+    public boolean hasReturnValue() {
+      return Type.getMethodType(desc).getReturnType() != Type.VOID_TYPE;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
There is an overall lack of tests regarding equals/hashcode of Instruction. Worth a dedicated commit or even PR.

Reply:
tests added